awx/main/tasks.py

```diff
@@ -2393,6 +2410,14 @@ def post_run_hook(self, instance, status):
             if status == 'successful' and instance.launch_type != 'sync':
                 self._update_dependent_inventories(instance, dependent_inventory_sources)
 
+        # Clean up the oauth token if we are project syncing
+        if instance.project.sync_assets:
+            try:
+                OAuth2AccessToken.objects.filter(user=instance.created_by_id, description='project sync {}'.format(instance.id)).delete()
```

Comment:
A pattern we usually use for this sort of thing is storing the reference to the token above:

https://github.com/ansible/awx/pull/7099/files#diff-9d4ea1dd908b35fb92eaede4bd10bb46R2080

...as `self.new_token = `

...and then down here, you can just `self.new_token.delete()`

Reply:
Fixed.